kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/KeyManager.java

```diff
@@ -21,18 +24,18 @@
 public interface KeyManager<K> {
 
     /**
-     * Asynchronously encrypt the given {@code recordRequests} using the current DEK for the given KEK, calling the given receiver for each encrypted record
+     * Asynchronously encrypt the given {@code records} using the current DEK for the given KEK returning a transformed MemoryRecords
```

Comment:
"a transformed MemoryRecords" makes it sound like MemoryRecords the original is mutated.  I don't think that's the desired contract we mean.

```suggestion
     * Asynchronously encrypt the given {@code records} using the current DEK for the given KEK returning a MemoryRecords object which will contain all records transformed according to the {@code encryptionScheme}.
```